Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -1182,6 +1182,14 @@ bool RenderElement::repaintAfterLayoutIfNeeded(const RenderLayerModelObject* rep
     LayoutRect newOutlineBox;
 
     bool fullRepaint = selfNeedsLayout();
+
+    if (!fullRepaint && style().hasBorderRadius()) {
+        auto oldRadius = style().getRoundedBorderFor(oldBounds).radii();
+        auto newRadius = style().getRoundedBorderFor(newBounds).radii();
+
+        fullRepaint = oldRadius != newRadius;
```

Comment:
I am bit confused. if there's a geometry change (oldBounds != newBounds) aren't we supposed to issue a repaint on the enclosing geometry a few lines below?

Reply:
Do you mean `mustRepaintBackgroundOrBorder(*this) && (newBounds != oldBounds || newOutlineBox != oldOutlineBox)` ?

It seems that `mustRepaintBackgroundOrBorder`  when verifying the border case just returns true   `if (renderer.style().hasBorder() && renderer.borderImageIsLoadedAndCanBeRendered())`


It doesn't check for differences on border-radius. Should we move the border-radius comparison to inside this function instead?